### ISSUE TITLE
Text border

### ISSUE THREAD

**Issue by BernhardWerner** [CindyJS/CindyJS#769]
I would like to be able to draw text with a border for better contrast. I.e., something like a modifier for `drawtext` that enables it and a few more for border size and colour.

**Comment by strobelm** [comment]
To take the considerations of @kortenkamp into account we could change 
the return value of drawtext to return the corners of the bounding box. 
Then the user could take care of their bounding boxes themselves.

**Comment by kortenkamp** [comment]
No. This is not possible - if you have multiple viewports then this is not defined.

And even worse: if you have a non-Euclidean viewport (not available yet in CindyJS, but in Cinderella) the rectangle would not be a rectangle.

**Comment by kortenkamp** [comment]
As a workaround, btw, you could use a `Text` object

**Comment by montaga** [comment]
I am not sure, whether @BernhardWerner  was refering to a rectengular border:
Sometimes I used the following dirty trick:
```
drawwithborder(pos, txt, color, bordercolor) := (
  forall([[1,0],[-1,0],[0,1],[0,-1]],o,
         drawtext(pos, txt, color->bordercolor, offset->o);        
        );
  drawtext(pos, txt, color->color);
);

drawwithborder((0,0), "Hello World", [1,1,1], [0,0,0]);
```
which gives
![image](https://user-images.githubusercontent.com/15124152/53796669-e7b8ad80-3f34-11e9-8a50-688533a971a3.png)

It should be possible to implement a modifier to that. There are some canvas operations rendering this directly (https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/strokeText)

**Comment by BernhardWerner** [comment]
That is good enough for me, @montaga, thank you. A modifier would be, of course, still desirable.

**Comment by BernhardWerner** [comment]
Just added a basic implementation for text outlines in the branch `text-outline`  . You can now write something like this:
`drawtext([0,0], "text text text", size -> 30, color -> (1,1,1), outlinewidth -> 10, outlinecolor -> (0,0,0));`
Also see the appropriate example file `text-outline.html`. Open problems/questions:
1. The default behaviour (no outline) when those modifiers aren't present isn't working yet.
2. Are there better names for these modifiers?
3. The outline doesn't play well with alpha values below 1.

While I work on 1 and 3, feel free to test it out.

**Comment by kortenkamp** [comment]
Thank you for this – for alpha values we should have a look at the original implementation of alpha in Cinderella. Actually, there are two ways of doing alpha, one is "real alpha", which changes the opaqueness, and one is "mixed alpha" which just blends the drawing color with the background color. 

It would be great to have a coherent implementation of both alpha types throughout, and a modifier for choosing which alpha you like…

**Comment by BernhardWerner** [comment]
> Thank you for this – for alpha values we should have a look at the original implementation of alpha in Cinderella. Actually, there are two ways of doing alpha, one is "real alpha", which changes the opaqueness, and one is "mixed alpha" which just blends the drawing color with the background color.
> 
> It would be great to have a coherent implementation of both alpha types throughout, and a modifier for choosing which alpha you like…

Should not be too difficult to implement these two alpha. I hope... 
I'll put it on my list.

**Comment by BernhardWerner** [comment]
A new version is online that also works, in principle, for TeX formulas.

**Comment by BernhardWerner** [comment]
I just saw that there are a few issues with how the modifiers are handled. Will look at it over the next days. Until then, please don't create a new release of CindyJS, @kortenkamp .

**Comment by kortenkamp** [comment]
I merged #926, but still `node make alltests` will complain. Can you check again?

**Comment by BernhardWerner** [comment]
Will take a look as soon as possible!
